bundles/org.openhab.binding.melcloud/src/main/java/org/openhab/binding/melcloud/internal/handler/MelCloudAccountHandler.java

```diff
@@ -0,0 +1,205 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.melcloud.internal.handler;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerService;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.melcloud.internal.api.MelCloudConnection;
+import org.openhab.binding.melcloud.internal.api.json.Device;
+import org.openhab.binding.melcloud.internal.api.json.DeviceStatus;
+import org.openhab.binding.melcloud.internal.config.AccountConfig;
+import org.openhab.binding.melcloud.internal.discovery.MelCloudDiscoveryService;
+import org.openhab.binding.melcloud.internal.exceptions.MelCloudCommException;
+import org.openhab.binding.melcloud.internal.exceptions.MelCloudLoginException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link MelCloudAccountHandler} is the handler for MELCloud API and connects it
+ * to the webservice.
+ *
+ * @author Luca Calcaterra - Initial contribution
+ * @author Pauli Anttila - Refactoring
+ * @author Wietse van Buitenen - Return all devices
+ */
+public class MelCloudAccountHandler extends BaseBridgeHandler {
+    private final Logger logger = LoggerFactory.getLogger(MelCloudAccountHandler.class);
+
+    private MelCloudConnection connection;
+    private List<Device> devices;
+    private ScheduledFuture<?> connectionCheckTask;
+    private AccountConfig config;
+    private boolean loginCredentialError;
+
+    public MelCloudAccountHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(MelCloudDiscoveryService.class);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing MELCloud account handler.");
+        config = getThing().getConfiguration().as(AccountConfig.class);
+        connection = new MelCloudConnection();
+        devices = Collections.emptyList();
+        loginCredentialError = false;
+        scheduler.execute(() -> {
+            try {
+                connect();
```

Comment:
Any reason not directly call `startConnectionCheck` and set start time from `30` to `0`. Doing a separate scheduled call to `connect` only seems to make it more complex?